CRM/Core/Payment/PayPalIPN.php

```diff
@@ -398,9 +398,7 @@ public function getInput(&$input, &$ids) {
        * The `payment_date` that Paypal sends back is in their timezone. Example return: 08:23:05 Jan 11, 2019 PST
        * Subsequently, we need to account for that, otherwise the recieve time will be incorrect for the local system
        */
-      $systemTimeZone = new DateTimeZone(CRM_Core_Config::singleton()->userSystem->getTimeZoneString());
-      $receiveDateTime->setTimezone($systemTimeZone);
-      $input['receive_date'] = $receiveDateTime->format('YmdHis');
+      $input['receive_date'] = CRM_Utils_Date::convertDateToLocalTime($receiveDateTime);
```

Comment:
(1) I like how the name of the extracted function adds more clarity/pithiness.

(2) Small nitpick: I did a double-take wondering how `PayPalIPN.php` is related to the PR topic. Since the method-extraction is a bit tangential to the overall theme of adding `civicrm_prevnext_cache.expiry_date`, it might've made sense to do it as a separate commit. (Not critical, though.)

Reply:
@seamuslee001 Agree with @totten here - let's break this into a separate PR as it's a good change but has nothing to do with the expiry_date - and put the `convertDateToLocalTime()` function there too - we could get that merged quickly and then finish reviewing this one.

Reply:
In fact I would argue it it an unrelated change since we don't convert dates when writing to a timestamp field - mysql handles that